topics/RDMA.md

```diff
@@ -0,0 +1,167 @@
+---
+title: "RDMA support"
+linkTitle: "RDMA support"
+weight: 1
+description: Valkey Over RDMA support
+aliases: [
+    /topics/RDMA,
+    /docs/manual/RDMA,
+    /docs/manual/RDMA.md
+]
+---
+
+Valkey supports the RDMA connection type as a module that requires compilation
+as a shared library and dynamic loading on demand.
+
+## Protocol
+
+Note that the protocol part defines the QP type RC (like TCP),
+communication commands, and payload exchange mechanism.
+This dependency is based solely on the RDMA (aka Infiniband) specification
+and is independent of both software (including the OS and user libraries)
+and hardware (including vendors and low-level transports).
+
+The Valkey Over RDMA protocol separates into control-plane
+(to exchange control message) and data-plane(to transfer the real payload for Valkey).
+
+### Control message
+
+For control message, use a fixed 32 bytes message which defines structures:
+```
+typedef struct ValkeyRdmaFeature {
+    /* defined as following Opcodes */
+    uint16_t opcode;
+    /* select features */
+    uint16_t select;
+    uint8_t rsvd[20];
+    /* feature bits */
+    uint64_t features;
+} ValkeyRdmaFeature;
+
+typedef struct ValkeyRdmaKeepalive {
+    /* defined as following Opcodes */
+    uint16_t opcode;
+    uint8_t rsvd[30];
+} ValkeyRdmaKeepalive;
+
+typedef struct ValkeyRdmaMemory {
+    /* defined as following Opcodes */
+    uint16_t opcode;
+    uint8_t rsvd[14];
+    /* address of a transfer buffer which is used to receive remote streaming data,
+     * aka 'RX buffer address'. The remote side should use this as 'TX buffer address' */
+    uint64_t addr;
+    /* length of the 'RX buffer' */
+    uint32_t length;
+    /* the RDMA remote key of 'RX buffer' */
+    uint32_t key;
+} ValkeyRdmaMemory;
+
+typedef union ValkeyRdmaCmd {
+    ValkeyRdmaFeature feature;
+    ValkeyRdmaKeepalive keepalive;
+    ValkeyRdmaMemory memory;
+} ValkeyRdmaCmd;
+```
+
+### Opcodes
+|Command| Value | Description |
+| :----: | :----: | :----: |
+| GetServerFeature   | 0 | required, get the features offered by Valkey server |
+| SetClientFeature   | 1 | required, negotiate features and set it to Valkey server |
+| Keepalive          | 2 | required, detect unexpected orphan connection |
+| RegisterXferMemory | 3 | required, tell the 'RX transfer buffer' information to the remote side, and the remote side uses this as 'TX transfer buffer' |
```

Comment:
Use backticks around command names to avoid spellcheck errors.

```suggestion
| `GetServerFeature`   | 0 | required, get the features offered by Valkey server |
| `SetClientFeature`   | 1 | required, negotiate features and set it to Valkey server |
| `Keepalive`          | 2 | required, detect unexpected orphan connection |
| `RegisterXferMemory` | 3 | required, tell the 'RX transfer buffer' information to the remote side, and the remote side uses this as 'TX transfer buffer' |
```